ldms/src/core/ldms.h

```diff
@@ -2787,14 +2787,38 @@ int ldms_set_unpublish(ldms_set_t set);
 extern void ldms_set_delete(ldms_set_t s);
 
 /**
- * \brief Free the set reference. The set will not be deleted.
+ * \brief Drop a reference obtained by the LDMS API
  *
- * Only the set handle \c s will be freed. The set content will not be deleted.
+ * The LDMS API that acquire a set reference are the following:
+ *   - ldms_set_by_name()
  *
- * \param s	The metric set handle
+ * \param s The metric set handle
  */
 void ldms_set_put(ldms_set_t s);
 
+/**
+ * @brief Take a reference on a metric set
+ *
+ * Used by applications to take a reference on a metric set. An
+ * application should use this interface if it caches a pointer
+ * to the set in a data strucure. Use ldms_set_put_ to drop a
+ * reference obtained with this interface
+ *
+ * @param set The set handle
+ * @return ldms_set_t The set handle
+ */
+ldms_set_t ldms_set_get_(ldms_set_t set);
```

Comment:
Maybe this should just be "ldms_set_get" and the following merged with the existing "ldms_set_put". As far as I can tell, the only reason these things ldms_set_put_ is different from ldms_set_put is to put a different string in the tracing. But if we really care about the string in the tracing then these generic ldms_set_get_/ldms_set_put_ functions should probably work more like ref_get/ref_put and allow a string an additional string option.